src/assets/wise5/components/label/label-student/label-student.component.ts

```diff
@@ -292,13 +292,27 @@ export class LabelStudent extends ComponentStudent {
       this.setStudentWork(componentState);
     } else if (this.component.hasConnectedComponent()) {
       this.handleConnectedComponents();
-      if (this.componentContent.labels != null) {
-        this.addLabelsToCanvas(componentContent.labels);
+      if (componentContent.labels != null) {
+        this.setStarterLabels(componentContent);
       }
     } else if (this.LabelService.componentStateIsSameAsStarter(componentState, componentContent)) {
       this.setStudentWork(componentState);
     } else if (componentState == null && componentContent.labels != null) {
-      this.addLabelsToCanvas(componentContent.labels);
+      this.setStarterLabels(componentContent);
+    }
+  }
+
+  private setStarterLabels(componentContent: any): void {
+    this.setIsStarterLabelTrue(componentContent.labels);
+    this.addLabelsToCanvas(componentContent.labels);
+  }
+
+  /**
+   * This is for old Label component content before we had the isStarterLabel field in the labels
+   */
```

Comment:
This comment is a bit confusing because the function can be called with labels that have isStarterLabel field already set to true (e.g. newly-authored content). Rephrase or remove?

Reply:
Revised comment is clearer, but I think it should be moved to just above the call to the ```setIsStarterLabelTrue()``` inside ```setStarterLabels()```, instead of above the ```setIsStarterLabelTrue()``` function definition.